### PR TITLE
[AMDGPU] Fix missing `IsExact` flag when expanding vector binary operator

### PR DESCRIPTION
**Pull request by shiltian**

### CODE REVIEW

**Comment by shiltian** [comment]
I'll update the test case in https://github.com/llvm/llvm-project/pull/83155 once that patch is landed to test this patch.

**Review by arsenm — CHANGES_REQUESTED** [review]
Missing test

**Comment by arsenm** [comment]
> I'll update the test case in #83155 once that patch is landed to test this patch.

The code is already tested, just add a case with exact

File: llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp [hunk 1]
```diff
@@ -1594,6 +1594,11 @@ bool AMDGPUCodeGenPrepareImpl::visitBinaryOperator(BinaryOperator &I) {
           }
         }
 
+        if (auto *NewEltI = dyn_cast<Instruction>(NewElt))
+          if (PossiblyExactOperator::isPossiblyExactOpcode(
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp:1598]: Can you use something simpler like `NewEltI->copyIRFlags(I)`?

**Comment by shiltian** [comment]
> > I'll update the test case in #83155 once that patch is landed to test this patch.
> 
> The code is already tested, just add a case with exact

Done.

**Review by jayfoad — APPROVED** [review]
LGTM

**Pull request merged**